### ISSUE TITLE
NoSuchElementException when deriving ConfigDescriptorProvider for case object

### ISSUE THREAD

**Issue by egast** [zio/zio-config#206]
When automatically deriving a `ConfigDescriptorProvider` for a `case object` a `java.util.NoSuchElementException: head of empty list` exception is thrown. The issue seems to be in the `ConfigDescriptorProvider.combine` method where it is trying to collect `ConfigDescriptorProvider`'s for all parameters in a `case class`. Since a `case object` does not have any parameters and the `head` function is called on this parameter list, it throws an exception. 

I have this issue with a configuration with that has an AutoOffsetStrategy (from zio-kafka) field.
```Scala
sealed trait AutoOffsetStrategy { self =>
    def toConfig = self match {
      case AutoOffsetStrategy.Earliest => "earliest"
      case AutoOffsetStrategy.Latest   => "latest"
      case AutoOffsetStrategy.None     => "none"
    }
  }

  object AutoOffsetStrategy {
    case object Earliest extends AutoOffsetStrategy
    case object Latest   extends AutoOffsetStrategy
    case object None     extends AutoOffsetStrategy
  }
```
Trace:
`Caused by: java.util.NoSuchElementException: head of empty list
	at scala.collection.immutable.Nil$.head(List.scala:592)
	at scala.collection.immutable.Nil$.head(List.scala:591)
	at zio.config.magnolia.ConfigDescriptorProvider$$anon$3.getDescription(ConfigDescriptorProvider.scala:78)
	at zio.config.magnolia.ConfigDescriptorProvider$$anon$4.getDescription(ConfigDescriptorProvider.scala:104)
	at zio.config.magnolia.ConfigDescriptorProvider$$anon$3.$anonfun$getDescription$2(ConfigDescriptorProvider.scala:58)
	at scala.collection.immutable.List.map(List.scala:223)
	at zio.config.magnolia.ConfigDescriptorProvider$$anon$3.getDescription(ConfigDescriptorProvider.scala:55)
	at zio.config.magnolia.ConfigDescriptorProvider$$anon$3.$anonfun$getDescription$2(ConfigDescriptorProvider.scala:58)
	at scala.collection.immutable.List.map(List.scala:219)
	at zio.config.magnolia.ConfigDescriptorProvider$$anon$3.getDescription(ConfigDescriptorProvider.scala:55)
	at zio.config.magnolia.ConfigDescriptorProvider$$anon$3.$anonfun$getDescription$2(ConfigDescriptorProvider.scala:58)
	at scala.collection.immutable.List.map(List.scala:219)
	at zio.config.magnolia.ConfigDescriptorProvider$$anon$3.getDescription(ConfigDescriptorProvider.scala:55)
	at zio.config.magnolia.ConfigDescriptorProvider$.description(ConfigDescriptorProvider.scala:139)`

**Comment by afsalthaj** [comment]
Yes, this is on my radar, and will be fixed quickly.
As soon as the experimental tag is removed from magnolia module, and the entire project goes to 1.0 release, we will be making sure there is no unsafe operations to get non-emptiness behaviour

**Comment by afsalthaj** [comment]
@egast  Thanks for reporting!

**Comment by iyerland** [comment]
I can work on this @afsalthaj

**Comment by afsalthaj** [comment]
@iyerland  Fantasitc. Looking forward.
There are a few design changes coming in, but it shouldn't matter to magnolia code base.
Ask me any help.

The only thing that kind of burns me with `case object` is, it will come with surprises.

Such as `lower case values in congig string` correspond to the `camel case` in case object definitions. I request you to stick on to least surprise principle, and ask me in discord for any help with this

**Comment by afsalthaj** [comment]
@egast 
If you are blocked by any chance, please use explicit instances of the case objects. We have been pretty busy with another PR, and hasn't got the time to look at the issue.

See below:

```scala
  sealed trait AutoOffsetStrategy

  object AutoOffsetStrategy {
    case object Earliest extends AutoOffsetStrategy
    case object Latest   extends AutoOffsetStrategy
    case object None     extends AutoOffsetStrategy
  }

  def instance[A](value: String)(f: => A): ConfigDescriptorProvider[A] =
    new ConfigDescriptorProvider[A] {
      override def getDescription(path: String): ConfigDescriptor[String, String, A] =
        string("a").xmapEither(str => if (str == value) Right(f) else Left(s"Not ${value}"), e => Right("e"))
    }

  implicit val x = instance("earliest")(Earliest)
  implicit val y = instance("latest")(Latest)
  implicit val z = instance("none")(AutoOffsetStrategy.None)

  val source = ConfigSource.fromMap(Map("a" -> "earliest"))

  read(description[AutoOffsetStrategy] from source
  // this should return `Earliest`
}

```

**Comment by afsalthaj** [comment]
In the next release, you won't be needing to write this boilerplate,  along with other changes. And thats 1.0

**Comment by afsalthaj** [comment]
Fixed in #236

**Comment by afsalthaj** [comment]
Fixed this in #236 

https://github.com/zio/zio-config/pull/236/commits/8241954b208752a7bea1c0692faf933742f99c0c

**Comment by afsalthaj** [comment]
Merged and released v1.0.0-RC14
